ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuationTest.kt

```diff
@@ -0,0 +1,172 @@
+package com.pinterest.ktlint.ruleset.standard.rules
+
+import com.pinterest.ktlint.test.KtLintAssertThat.Companion.assertThatRule
+import com.pinterest.ktlint.test.LintViolation
+import org.junit.jupiter.api.Test
+
+class ChainMethodContinuationTest {
+    private val chainMethodContinuationAssertThat = assertThatRule { ChainMethodContinuation() }
+
+    @Test
+    fun `Given method chain which is correctly wrapped in single line`() {
+        val code =
+            """
+            val foo = listOf(1, 2, 3).filter { it > 2 }!!.takeIf { it.count() > 100 }.map { it * it }?.sum()!!
+            """.trimIndent()
+        chainMethodContinuationAssertThat(code).hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given method chain which is incorrectly wrapped`() {
```

Comment:
I prefer to have smaller tests when possible. I think that this test should be split into 3 (one test for operators `.`, `?.` and `!!.`.

Reply:
Made this change. Let me know after reviewing the latest changes more splitting is required for code readability